common/kernel/hashlib.h

```diff
@@ -818,6 +828,11 @@ template <typename K, typename OPS> class pool
         const_iterator(const pool *ptr, int index) : ptr(ptr), index(index) {}
 
       public:
+        using iterator_category = std::forward_iterator_tag;
+        using value_type = K;
+        using difference_type = std::ptrdiff_t;
+        using pointer = K*;
```

Comment:
I think this should be `const K*` and `'const K&`

Reply:
Fixed, although my understanding based on the defaults of `std::iterator` is that non-`const` was the behaviour here, and I didn't want to change that without your approval.